app/src/main/java/com/conupods/OutdoorMaps/MapInitializer.java

```diff
@@ -7,22 +7,116 @@
 import android.widget.EditText;
 import android.widget.TextView;
 
+import com.conupods.IndoorMaps.IndoorBuildingOverlays;
 import com.conupods.R;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.LatLngBounds;
+
+import java.util.ArrayList;
+import java.util.List;
 
 import static android.view.inputmethod.EditorInfo.IME_ACTION_DONE;
 import static android.view.inputmethod.EditorInfo.IME_ACTION_SEARCH;
 
-public class MapInitializer {
+public class MapInitializer{
+
+    public static final LatLng CENTER_OF_HALL = new LatLng(  45.49728190486448,  	-73.57892364263535);
+    public static final LatLng CENTER_OF_JMSB = new LatLng(45.49524950613837, -73.57895582914352);
+    public static final LatLng CENTER_OF_LOY_CC = new LatLng(45.45824552786007,    -73.64033281803131);
+    public static final LatLng CENTER_OF_LOY_VL = new LatLng( 45.459086497919344,     -73.63828897476196);
+    public static final double mZoomLevel = 18.3;
+
 
-    CameraController mCameraController;
+    private CameraController mCameraController;
+    private IndoorBuildingOverlays mIndoorBuildingOverlays;
+    private OutdoorBuildingOverlays mOutdoorBuildingOverlays;
+    private GoogleMap mMap;
     private static final String TAG = "MapInitializer";
 
-    public MapInitializer(CameraController CameraController) {
-        mCameraController = CameraController;
+    List<Button> buttonsHALL = new ArrayList<Button>();
+    List<Button> buttonsJMSB = new ArrayList<Button>();
+    List<Button> buttonsLOYCC = new ArrayList<Button>();
+
+    public MapInitializer(CameraController cameraController, IndoorBuildingOverlays indoorBuildingOverlays, OutdoorBuildingOverlays outdoorBuildingOverlays, GoogleMap map) {
+        mCameraController = cameraController;
+        mIndoorBuildingOverlays = indoorBuildingOverlays;
+        mOutdoorBuildingOverlays = outdoorBuildingOverlays;
+        mMap = map;
+    }
+
+    public void onCameraChange(){
+
+        mMap.setOnCameraIdleListener(()->{
+
+            LatLngBounds bounds = mMap.getProjection().getVisibleRegion().latLngBounds;
+
+            if (mMap.getCameraPosition().zoom > mZoomLevel) {
```

Comment:
u can use chain of responsibility here

Reply:
Decided not to use chain of responsibility here.